src/viam/sdk/registry/registry.hpp

```diff
@@ -11,86 +11,56 @@
-    /// @param name The name of the resource.
-    /// @param channel A channel connected to the client.
-    /// @return A `shared_ptr` to the resource client.
-    virtual std::shared_ptr<Resource> create_rpc_client(std::string name,
-                                                        std::shared_ptr<grpc::Channel> channel) = 0;
+    /// @brief Returns a reference to the `ResourceServerRegistration`'s service descriptor.
+    const google::protobuf::ServiceDescriptor* service_descriptor() const;
 
-    ResourceRegistration(const google::protobuf::ServiceDescriptor* service_descriptor)
+    ResourceServerRegistration(const google::protobuf::ServiceDescriptor* service_descriptor)
         : service_descriptor_(service_descriptor){};
 
    private:
     const google::protobuf::ServiceDescriptor* service_descriptor_;
 };
 
-// TODO(RSDK-3030): Potentially make ResourceRegistration2 the one and only
-// form of resource registration.
-template <typename ResourceClientT,
-          typename ResourceServerT,
-          typename ProtoServiceT,
-          typename ResourceRegT>
-class ResourceRegistration2 : public ResourceRegistration {
+/// @class ResourceClientRegistration registry.hpp "registry/registry.hpp"
+/// @brief Defines registered `Resource` client creation functionality.
+class ResourceClientRegistration {
    public:
-    using ResourceRegistration::ResourceRegistration;
-    std::shared_ptr<ResourceServer> create_resource_server(std::shared_ptr<ResourceManager> manager,
-                                                           Server& server) override {
-        auto rs = std::make_shared<ResourceServerT>(manager);
-        server.register_service(rs.get());
-        return rs;
-    }
+    virtual ~ResourceClientRegistration();
 
-    std::shared_ptr<Resource> create_rpc_client(std::string name,
-                                                std::shared_ptr<grpc::Channel> chan) override {
-        return std::make_shared<ResourceClientT>(std::move(name), std::move(chan));
-    }
+    /// @brief Create gRPC client to a resource.
+    /// @param name The name of the resource.
+    /// @param channel A channel connected to the client.
+    /// @return A `shared_ptr` to the resource client.
+    virtual std::shared_ptr<Resource> create_rpc_client(
+        std::string name, std::shared_ptr<grpc::Channel> channel) const = 0;
 
-    static std::shared_ptr<ResourceRegistration> resource_registration() {
-        const google::protobuf::DescriptorPool* p =
-            google::protobuf::DescriptorPool::generated_pool();
-        const google::protobuf::ServiceDescriptor* sd =
-            p->FindServiceByName(ProtoServiceT::service_full_name());
-        if (!sd) {
-            throw std::runtime_error("Unable to get service descriptor");
-        }
-        return std::make_shared<ResourceRegT>(sd);
-    }
+    ResourceClientRegistration() = default;
 };
 
+// TODO(RSDK-3030): instead of std::functions, consider making these functions
```

Comment:
Should we use a different ticket number if this is still a `TODO`?

Reply:
Ooh yes, good catch! Updated.